src/cozy/react/components/AvailablePlatforms.jsx

```diff
@@ -30,15 +30,23 @@ export const InstallNativeAppButton = props => {
     setSmartphoneModalOpened(false)
   }, [setSmartphoneModalOpened])
 
+  const onClick = () => {    
+    if(mobileOS === null) {
+      handleOpenModal()
+    } else {
+      props.onClick?.()
```

Comment:
⚠️ this method is not awaited, but the passed prop is `setVaultInstalled` which is async and does a HTTP query.

So I supposed you may have a race condition between the `client.query()`/`client.save()` code and the `location.href` navigation.

Reply:
Nice catch! Thank you!